lib/sinon/util/fake_server.js

```diff
@@ -94,7 +95,21 @@ if (typeof sinon == "undefined") {
 
                 return server;
             },
-
+            configure: function (config) {
+                var whitelist = {
+                    "autoRespond": true,
+                    "autoRespondAfter": true,
+                    "respondImmediately": true,
+                    "fakeHTTPMethods": true
+                },
+                setting;
+                config = config || {};
+                for (setting in config) {
```

Comment:
Sinon.JS needs to be very defensive and shouldn't fail even when end users do questionable things like extending `Object.prototype`.

With that in mind, you'll need to protect `for..in` with `hasOwnProperty`

The [PR to add ESLint for validation](https://github.com/cjohansen/Sinon.JS/pull/775) would have caught this, but is not merged yet.